### PR TITLE
Update Slice with dynamic input & optional input steps

### PR DESCRIPTION
**Pull request by BowenBao**
* Replacing Slice with DynamicSlice: now `starts`, `ends` and `axes` are inputs, instead of attributes.
* Remove DynamicSlice from Experiment Ops.
* Add optional input `steps`.

### CODE REVIEW

File: onnx/defs/tensor/defs.cc [hunk 1]
```diff
@@ -471,55 +474,96 @@ Example 2:
...
-            "Starting indices of corresponding axis in `axes`",
-            AttributeProto::INTS)
-        .Attr(
-            "ends",
-            "Ending indices (exclusive) of corresponding axis in axes`",
-            AttributeProto::INTS)
+        .Input(1, "starts", "1-D tensor of starting indices of corresponding axis in `axes`", "Tind")
+        .Input(2, "ends", "1-D tensor of ending indices (exclusive) of corresponding axis in `axes`", "Tind")
+        .Input(3, "axes", "1-D tensor of axes that `starts` and `ends` apply to.", "Tind", OpSchema::Optional)
+        .Input(4, "steps", "1-D tensor of slice step of corresponding axis in `axes`. Default to 1. ", "Tind", OpSchema::Optional)
```
Comment (linkerzhang) [inline onnx/defs/tensor/defs.cc]: so the "steps" is used to specify the axis order/step of slicing? if that's the case, can we just use/follow the "axes" indices order please?

Comment (BowenBao) [reply onnx/defs/tensor/defs.cc]: I'm not sure I understand the question correctly. Is your question suggesting that for a given axis `j = axes[i]`, `output[i]` should be computed based on `starts[j]`, `ends[j]`, `steps[j]`? If that's the case then yes that is the current design, I'll update the description to be more precise.

**Comment by BowenBao** [comment]
@linkerzhang part of the CI is failing on various caffe2 & pytorch test cases & models using the old slice spec(<opset9). What is the recommended way of working around that?

**Comment by houseroad** [comment]
This change looks good to me. But we should prepare for it.

**Comment by spandantiwari** [comment]
@houseroad - all the tests are passing. Shall we go ahead and merge it now?

**Comment by houseroad** [comment]
Yeah, it should be fine to merge.

**Comment by ayermolo** [comment]
@BowenBao So to clarify. This is for opset 10. For opset < 10 starts/ends is still part of the attributes?

**Comment by BowenBao** [comment]
> @BowenBao So to clarify. This is for opset 10. For opset < 10 starts/ends is still part of the attributes?

yes.

**Pull request merged**